src/mp/delta.cpp

```diff
@@ -0,0 +1,139 @@
+/*
+ * Copyright (c) 2020 The reone project contributors
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+#include <cstdint>
+#include <array>
+#include <vector>
+#include <tuple>
+
+#include "delta.h"
+#include "objects.h"
+
+using namespace std;
+
+namespace reone {
+
+namespace mp {
+
+template <class TuplePtr, size_t ... Is>
+auto getSerImpl(TuplePtr &&seqPtr, integer_sequence<size_t, Is...>) {
+    return make_tuple(*(get<Is>(seqPtr))...);
+}
+
+template <class T>
+auto getSerial(T &obj) {
+    return getSerImpl(getSerialPointers(obj), make_index_sequence<tuple_size<decltype(getSerialPointers(obj))>::value>());
+}
+
+/* helper functions */
```

Comment:
Let's use C++-style comments for code folding and implementation notes, and Javadoc-style comments for documentation. See [https://github.com/seedhartha/reone/wiki/Code-Style#comments](https://github.com/seedhartha/reone/wiki/Code-Style#comments).